.evergreen/create-expansions.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Get the current unique version of this checkout.
+if [[ "$is_patch" = true ]]; then
+  CURRENT_VERSION=$(git describe)-patch-${version_id}
+else
+  CURRENT_VERSION=latest
+fi
+
+DRIVERS_TOOLS="$(pwd)/../drivers-tools"
+
+# Python has cygwin path problems on Windows. Detect prospective mongo-orchestration home
+# directory.
+if [[ "${OS}" = "Windows_NT" ]]; then
+  export DRIVERS_TOOLS=$(cygpath -m $DRIVERS_TOOLS)
+fi
+
+MONGO_ORCHESTRATION_HOME="${DRIVERS_TOOLS}/.evergreen/orchestration"
+MONGODB_BINARIES="${DRIVERS_TOOLS}/mongodb/bin"
+PROJECT_DIRECTORY="$(pwd)"
+TMPDIR="${MONGO_ORCHESTRATION_HOME}/db"
+PATH="${MONGODB_BINARIES}:${PATH}"
+PROJECT="${project}"
+
+cat <<EOT >expansion.yml
+CURRENT_VERSION: "${CURRENT_VERSION}"
+DRIVERS_TOOLS: "${DRIVERS_TOOLS}"
+MONGO_ORCHESTRATION_HOME: "${MONGO_ORCHESTRATION_HOME}"
+MONGODB_BINARIES: "${MONGODB_BINARIES}"
+PROJECT_DIRECTORY: "${PROJECT_DIRECTORY}"
+TMPDIR: "${TMPDIR}"
+PATH: "${PATH}"
```

Comment:
so this part aims to output various env variables to stdout for user's inspection purposes, as demonstrated below:
```
cat <<EOT >expansion.yml
... ... 
EOT
# Print the expansion file created.
cat expansion.yml
```

so the `expansion.yml` is a temporary file only for user inspection purposes. The lines you mentioned is the real file content user will see, so it grabs env variable values and inserts them within the `${}` placeholder.

`PREPARE_SHELL` is always a special env variable which is kind of bash script header. Basically it will configure some common shell behavour, plus exporting a bunch of env variables.

The current pattern is whenever we run some bash script, we need to include `$PREPARE_SHELL` as the first step, e.g.:

```
    - command: shell.exec
      params:
        script: |
          ${PREPARE_SHELL}
          echo "{ \"releases\": { \"default\": \"$MONGODB_BINARIES\" }}" > $MONGO_ORCHESTRATION_HOME/orchestration.config
```

Unfortunately it is shell-specific so it won't work with subprocess (subprocess will rely on `include_expansions_in_env` to configure env variables. So it is an ugly legacy pattern and it conflicts with the `using subprocess instead of shell.exec` best practice.

In Rust config, `subprocess` was used as much as possible as long as the `${PREPARE_SHELL}` is not needed.

Reply:
well, seems the file is not temporary output purposes only. It is used in the `create expansions` task as the last step. I must honestly admit I have no idea what is its purpose and how it is gonna be used.

Reply:
From my understanding, when we create expansions file, we also create a `PREPARE_SHELL` expansion entry as part of it, which is used whenever we run `shell.exec` task (for simple interactive shell processing like deleting some file, using subprocess seems overkill and clumsy). 

It includes some common env variables which might not be used all the time in a `shell.exec` task but good to know for troubleshooting purposes (their values will be logged somewhere).

Reply:
Resolving this (there are 3 comments that end up being about PREPARE_SHELL), and asking for either docs or a ticket in [this comment](https://github.com/mongodb/mongo-hibernate/pull/9#discussion_r1868288440).